compiler/resolution/wrappers.cpp

```diff
@@ -1371,7 +1372,7 @@ static void addArgCoercion(FnSymbol*  fn,
         !fn->hasFlag(FLAG_INIT_COPY_FN)) {
 
       bool isConstCopy = formal->intent == INTENT_CONST_IN ||
-                         formal->intent == INTENT_CONST_IN ||
+                         formal->originalIntent == INTENT_CONST_IN ||
```

Comment:
This change feels workaround-y. Is it really required? Why is it needed? Is it related to the default initializer change?

Reply:
This was there before, but we had `intent` twice instead of having `originalIntent` for one of them, so this change is not directly related to what this PR is aiming to do.

That being said, this didn't cause any issues so far, and maybe you are questioning the general need for this here?

Reply:
Ah, sorry. Yeah, let's keep this change if it's testing. It seems like a bug fix. I hadn't noticed the duplicate check.